feat: update with encrypted database.

hello-world-bfree opened the pull request:
This ensure the data always stays with the repo while simultaneously preventing the candidate from accessing the data too early. We'll provide them the password at the time of the interview.

hello-world-bfree committed:
Update with encrypted database.

dspainte commented on README.md:
```diff
@@ -7,6 +7,7 @@ General Requirements:
 - Some stub code has been provided, you do not have to use it or even use Javascript if you do not wish
```
May want to include a note here about making sure they have run `npm run setup` in order to set up the database + docker container ahead of time?  That might be easy to miss, but if they don't have docker running, they'll have to download it, etc and wouldn't want to have them doing that on interview time

hello-world-bfree replied:
No problem! I mention it in the `database access` bullet two down, but I think adding it as an explicit TODO is important. Thanks!

dspainte reviewed (approved):
This is a huge improvement, thanks!  

One thing is that the first time I ran `npm run setup` on my local I got this error:

<img width="486" height="575" alt="Screenshot 2025-10-01 at 9 26 36 AM" src="https://github.com/user-attachments/assets/5630358f-eb20-4190-b540-ae73a9423b49" />

When I ran it the second time it worked fine?  But may want to fix that so candidates don't get confused

hello-world-bfree commented:
> One thing is that the first time I ran `npm run setup` on my local I got this error: ...
> When I ran it the second time it worked fine?

This looks like it happened because you had an unrelated container named `database`. So that was deleted, but the script assumes because you had the `database` container that it was from the `interview-db` image so it tries to remove that as well.

It does probably indicate that `database` is too generic of a container name 🤦 hahah I'll change it to something more unique to avoid this! Thanks!

dspainte commented:
> > One thing is that the first time I ran `npm run setup` on my local I got this error: ...
> > When I ran it the second time it worked fine?
> 
> This looks like it happened because you had an unrelated container named `database`. So that was deleted, but the script assumes because you had the `database` container that it was from the `interview-db` image so it tries to remove that as well.
> 
> It does probably indicate that `database` is too generic of a container name 🤦 hahah I'll change it to something more unique to avoid this! Thanks!

Yeah I think that was exactly it - `database` also conflicts with our backend db container name 😆  Changing it sounds good!

hello-world-bfree committed:
update to add explicit documentation to run
> interview-backend@1.0.0 setup
> npm install && npm run db:start

up to date, audited 41 packages in 862ms

4 packages are looking for funding
  run `npm fund` for details

4 vulnerabilities (2 moderate, 1 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> interview-backend@1.0.0 db:start
> sh start_database.sh

=======================================
  Interview Database Setup
=======================================

Container 'my-interview-db' already exists.

Exiting. Please remove the container manually or use a different name.. Make port something more likely not to have a collision. Make the container name more unique to avoid a collision.

hello-world-bfree commented:
@dspainte There was also a similar issue to the container name with the port. In the event a candidate is working with mysql for their current role or a project they would probably be using port 3306 causing a port allocation error. So I updated that as well. Along with the documentation.
